cli/pcluster/utils.py

```diff
@@ -602,6 +602,23 @@ def get_master_ip_and_username(cluster_name):
     return master_ip, username
 
 
+def get_supported_archs_for_inst_type(instance_type):
+    """Get a list of architectures supported for the given instance type."""
+    try:
+        ec2_client = boto3.client("ec2")
+        instance_info = ec2_client.describe_instance_types(InstanceTypes=[instance_type]).get("InstanceTypes")[0]
+    except ClientError as e:
```

Comment:
Let's catch the KeyError too

Reply:
Those two lines cannot cause a KeyError. describe_instance_types raises a ClientError if the specified instance_type doesn't exist., which is handled If it does, then "InstanceTypes" will exist in the returned dict and it will have an entry.